src/basic/log.h

```diff
@@ -375,34 +375,50 @@ typedef struct LogRateLimit {
         RateLimit ratelimit;
 } LogRateLimit;
 
-#define log_ratelimit_internal(_level, _error, _format, _file, _line, _func, ...)        \
+#define log_ratelimit_internal(_level, _error, _ratelimit, _format, _file, _line, _func, ...)        \
```

Comment:
This makes multiple logs grouped with the same ratelimit, even if their messages or contexts are quite different.
And the automatically appended `" (Dropped %u similar message(s))"` is quite misleading.

Reply:
> This makes multiple logs grouped with the same ratelimit, even if their messages or contexts are quite different.
And the automatically appended " (Dropped %u similar message(s))" is quite misleading.

This was already the case before this PR. Note that the ratelimit instance is unique for each call to log_ratelimit_internal() because it's copied into `LogRateLimit`. So a ratelimit is only reused for the same format string, which can be completed with different format arguments, but it's still the same log message and will look similar to the other log messages generated from the same format string, so they should be similar?

Reply:
Ah, indeed. That's my misunderstanding. Each log message is counted separately, but only its burst parameters are shared. Sorry for the noise.